rust/core-lib/src/event_context.rs

```diff
@@ -954,14 +953,29 @@ mod tests {
         ctx.do_edit(EditNotification::DecreaseNumber);
         assert_eq!(harness.debug_render(),"this is a 999| text example");
 
-        // Number Wrapping
+        // invalid numbers
         ctx.do_edit(EditNotification::MoveToEndOfDocument);
         ctx.do_edit(EditNotification::DeleteToBeginningOfLine);
-        ctx.do_edit(EditNotification::Insert { chars: format!("{}", i128::max_value()) });
-        ctx.do_edit(EditNotification::Gesture { line: 0, col: 11, ty: PointSelect });
-        ctx.do_edit(EditNotification::MoveRightAndModifySelection);
+        ctx.do_edit(EditNotification::Insert { chars: "10_000".into() });
+        ctx.do_edit(EditNotification::MoveToEndOfDocument);
+        ctx.do_edit(EditNotification::IncreaseNumber);
+        assert_eq!(harness.debug_render(), "10_000|");
+
+        // decimals are kinda accounted for (i.e. 4.55 becomes 4.56 (good), but 4.99 becomes 4.100 (bad)
```

Comment:
this is okay, we're never going to handle floating point 'correctly', and just consistently treating anything we handle as an integer feels okay to me.